python/ray/_raylet.pyx

```diff
@@ -1103,32 +1126,40 @@ cdef class CoreWorker:
         object_already_exists = self._create_put_buffer(
             metadata, total_bytes, object_ref,
             ObjectRefsToVector(serialized_object.contained_object_refs),
-            &c_object_id, &data, True)
+            &c_object_id, &data, True, owner_address)
 
         if not object_already_exists:
             if total_bytes > 0:
                 (<SerializedObject>serialized_object).write_to(
                     Buffer.make(data))
             if self.is_local_mode or (put_small_object_in_memory_store
                and <int64_t>total_bytes < put_threshold):
+                if owner_address is not None:
```

Comment:
I feel a little bit worry about this one. If the owner is set, should we always put it into the plasma store?

Reply:
I have thought about this question but there is no answer. The main reasons for doing so are as follows:
1. It seems that `put_small_object_in_memory_store` is false for a long time, basically, no one uses it
2. Put the data directly to the owner seems to be an efficiency problem, I didn't think about how to do it.

Reply:
I remember even we put it into memory, it'll be promoted to plasma store since it's borrowed by other nodes.

Reply:
Ok, I think I got the point here. Basically, it's for put only and by default, it looks like we don't do the small object optimization for put.

As a separate issue here, I'm wondering whether it's necessary to enable it by default. @stephanie-wang  do you have some suggestions?

Reply:
I think this is okay for now. It's being worked on in #13621, but we can address this case in a separate PR.